### PR TITLE
`TextField` house keeping 🧹 

### LINKED ISSUE #16642: 🧹 [UI House Keeping] `TextField`
### Description
Ensure that `TextField` adheres to all of the following conventions and standards

- [x] Has a `className` prop and the PropType descriptions are all the same
- [x] Prop table in MDX docs have the "Accepts all Box component props" description and link
- [x] We are consistent when using the same prop names like `size` and are suggesting the use of the generalized `design-system.js` constants e.g. `SIZES` as the primary option but noting the component consts in the documentation and using them for propType validation and storybook controls only
- [x] Standardize all similar prop names for images `imgSrc`, `imgAlt`(html element + attribute) (needs audit)
- [x] We have a story for each component prop and we use the prop name verbatim e.g. `size` prop would be `export const Size = (args) => (`
- [x] We have the accompanying documentation for each component prop and we use the prop name verbatim e.g. `size` prop would be `### Size`
- [x] Are multiple props stories allowed? e.g. `Color, Background Color And Border Color` story in `base-avatar` - [ ] yes when it makes sense to
- [x] All Base components follow the suffix convention e.g. `ButtonBase`
- [x] All Base component MDX documentation have the base component notification at the top
- [x] Add `mm-` prefix to all classNames
- [x] className is kebab case version of the component name
- [x] Spread base components props and reduce duplication of props when props aren't being changed and remain the same for both variant and base components
- [x] Add component to root `index.js` file in component-library
- [ ] Add locals for any default text I18nContext as default context 
- [x] Add any "to dos" with a `// TODO:` comment so we can search for them at a later date e.g. blocking components etc
- [x] Add snapshot testing
- [x] Add pixel values to propType descriptions if we use abstracted prop types that relate to pixel values e.g. `SIZE.MD (32px)`
- [x] Each prop section in the MDX docs should have: a heading, a description, a story and an example code snipped

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## Explanation

Updating `TextField` to include the most up to date standards and conventions for the component-library components listed in the issue and description below

* Fixes #16642  

- [x] Has a `className` prop and the PropType descriptions are all the same
- [x] Prop table in MDX docs have the "Accepts all Box component props" description and link
- [x] We are consistent when using the same prop names like `size` and are suggesting the use of the generalized `design-system.js` constants e.g. `SIZES` as the primary option but noting the component consts in the documentation and using them for propType validation and storybook controls only
- [x] Standardize all similar prop names for images `src`
- [x] We have a story for each component prop and we use the prop name verbatim e.g. `size` prop would be `export const Size = (args) => (`
- [x] We have the accompanying documentation for each component prop and we use the prop name verbatim e.g. `size` prop would be `### Size`
- [x] Are multiple props stories allowed? e.g. `Color, Background Color And Border Color` story in `base-avatar` - [ ] yes when it makes sense to
- [x] All Base components follow the suffix convention e.g. `ButtonBase`
- [x] All Base component MDX documentation have the base component notification at the top
- [x] Add `mm-` prefix to all classNames
- [x] className is kebab case version of the component name
- [x] Spread base components props and reduce duplication of props when props aren't being changed and remain the same for both variant and base components
- [x] Add component to root `index.js` file in component-library
- [ ] Add locals for any default text I18nContext as default context **Leaving this for now it need help with localization I'll create another issue**
- [x] Add any "to dos" with a `// TODO:` comment so we can search for them at a later date e.g. blocking components etc
- [x] Add snapshot testing
- [x] Add pixel values to propType descriptions if we use abstracted prop types that relate to pixel values e.g. `SIZE.MD (32px)`
- [x] Each prop section in the MDX docs should have: a heading, a description, a story and an example code snipped

### Before

https://user-images.githubusercontent.com/8112138/203663513-a158d920-4875-433c-9ec8-52775a801d4e.mov

### After

https://user-images.githubusercontent.com/8112138/203663541-9f2088ec-16d2-4e17-b2b2-80907a3aaa48.mov

## Manual Testing Steps

- Go to latest build of Storybook on this PR
- Search `TextField` in the search bar
- Check stories, controls and docs

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/component-library/component-library-components.scss [hunk 1]
```diff
@@ -23,7 +23,6 @@
 // Molecules
 @import 'picker-network/picker-network';
 @import 'tag-url/tag-url';
-@import 'text-field/text-field';
 @import 'text-field-base/text-field-base';
 @import 'text-field-search/text-field-search';
 @import 'form-text-field/form-text-field';
```
Comment (georgewrmarshall) [inline ui/components/component-library/component-library-components.scss:28]: No longer needed now that the button styles have been removed see `ui/components/component-library/text-field/text-field.scss`

File: ui/components/component-library/text-field/README.mdx [hunk 2]
```diff
@@ -12,15 +14,20 @@ The `TextField` component lets users enter and edit text as well as adding a sho
...
-The `TextField` accepts all props below as well as all [Box](/docs/ui-components-ui-box-box-stories-js--default-story#props) and [TextFieldBase](/docs/ui-components-component-library-text-field-base-text-field-base-stories-js--default-story#props) component props
+The `TextField` accepts all props below as well as all [Box](/docs/ui-components-ui-box-box-stories-js--default-story#props) component props
 
 <ArgsTable of={TextField} />
 
+`TextField` accepts all [TextFieldBase](/docs/ui-components-component-library-text-field-base-text-field-base-stories-js--default-story#props)
+component props
+
+<ArgsTable of={TextFieldBase} />
+
```
Comment (georgewrmarshall) [inline ui/components/component-library/text-field/README.mdx:25]: Adding child prop table 🧒

File: ui/components/component-library/text-field/text-field.scss [hunk 3]
```diff
@@ -1,10 +0,0 @@
-.mm-text-field {
-  // TOD: remove most of these styles when replaced by ButtonIcon
-  &__button-clear {
-    height: 24px;
-    width: 24px;
-    max-width: 24px;
-    flex: 0 0 24px;
-    margin-left: -24px;
-  }
-}
```
Comment (georgewrmarshall) [inline ui/components/component-library/text-field/text-field.scss:10]: Button styles are no longer needed and there are no other styles so we can remove. Still keeping class name as an identifier

File: ui/components/component-library/text-field/text-field.stories.js [hunk 4]
```diff
@@ -209,13 +209,6 @@ ShowClearButton.args = {
   showClearButton: true,
 };
 
-export const ClearButtonOnClick = Template.bind({});
-
-ShowClearButton.args = {
-  placeholder: 'Enter text to show clear',
-  showClearButton: true,
-};
```
Comment (georgewrmarshall) [inline ui/components/component-library/text-field/text-field.stories.js:217]: Removing unused story. I've bundled the `clearButtonOnClick` with the `showClearButton` story as they are both dependencies of each other

**Review by NidhiKJha — APPROVED** [review]

**Review by garrettbear — APPROVED** [review]

**Pull request merged**